Adding Doctor and Pharmacist Professions

theyareclose opened the pull request:
<!-- HOW TO USE: Under each "#### Heading" below, enter information relevant to your pull request.
Leave the headings unless they don't apply to your PR.

Please read carefully.
Once a pull request is submitted, automatic stylistic and consistency checks will be performed on the PR's changes.
The results can be either seen under the "Files changed" section of a PR or in the check's details.

Rules for suggested pull requests:
- If possible, limit yourself to small changes, 500 strings at max. Exceptions are adding or changing maps, and changes, that won't work unless they are done in a single run (even then there can be ways) - violating it puts a lot of unnecessary work on our merge team.
- Do not scope creep. If you make a pull request "Add new gun", please do not make anything more than adding the gun and following changes, like changing the stats of the gun, removing other guns from itemgroups or tweaking zombie horse stats - violating it makes future search and debugging stuff much harder, since PR name is not related to what is changed in the game. "Who the hell removed the quest item from drop in location X in PR, that adds a new plushie" - this may be a quote from a person who was affected by scope creep
- Do not make omnibus PRs. Meaning do not make a single PR, that fixes ten different, not related issues, at once, even if they are all one string - same as previously mentioned scope creep, it doesn't help to search the changes when debugging, despite all power of git blame tool

NOTE: Please grant permission for repository maintainers to edit your PR.  It is EXTREMELY common for PRs to be held up due to trivial changes being requested and the author being unavailable to make them. -->

#### Summary
Content "This adds a Doctor and Pharmacist profession. It also generalizes the "MD" trait description to include more medical professionals."
<!-- This section should consist of exactly one line, edit the one above.
1. Replace the word "Category" with one of these words: Features, Content, Interface, Mods, Balance, Bugfixes, Performance, Infrastructure, Build, I18N.
2. Replace the text inside the quotes with a brief description of your changes.
Or if you don't want a changelog entry, replace the whole line with just the word "None" (with no quotes).
Examples:
1. None
2. Features "In-game Armor sprite change"
3. Interface "Show crafting failure chances in the crafting interface"
4. Infrastructure "JSON-ize slot machines"
5. Bugfixes "Crafting GUI: show how much recipe makes for non-charge items"
For more on the meaning of each category, see:
https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/CHANGELOG_GUIDELINES.md
If approved and merged, your summary will be added to the project changelog:
https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/changelog.txt -->

#### Purpose of change
Previously, there was only the option to be a Medical Resident or Combat Medic, but not just a normal Doctor or Pharmacist. Both are fairly common professions, so this adds them to the game. The previous description of the medical degree trait, "MD" was very specific to the Medical Resident profession, so this generalizes it for use with more medical professions. Finally, I added Introduction to Chemistry to the Medical Resident as having a basic knowledge of chemistry is a typical requirement for medical school in the US.
<!-- With a few sentences, describe your reasons for making this change.
If it relates to an existing issue, you can link it with a # followed by the GitHub issue number, like #1234.
When you submit a pull request that completely resolves an issue, use [Github's closing keywords](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests#linking-a-pull-request-to-an-issue)
to automatically close the issue once your pull request is merged.
If there is no related issue, explain here what issue, feature, or other concern you are addressing.  If this is a bugfix, include steps to reproduce the original bug, so your fix can be verified. -->

#### Describe the solution
I added the Doctor and Pharmacist profession, edited the description for the MD trait, and added the Introduction to Chemistry proficiency to the Medical Resident. I achievement-gated the Doctor and Pharmacist behind the "Find a hospital" achievement, as that seemed fun and appropriate.
<!-- How does the feature work, or how does this fix a bug?  The easier you make your solution to understand, the faster it can get merged. -->

#### Describe alternatives you've considered
I considered adding Organic Chemistry as well for Medical Resident and Doctor as it's a common requirement, but I figured Physiology would cover the basics like the Krebs Cycle, etc. which is good enough. I considered Principles of Chemical Synthesis and Pharmaceutical Production for the Pharmacist, but Pharmacists don't really create drugs (outside of simple compounding), and I think those proficiencies are meant for actual Chemists.
<!-- Explain any alternative solutions, different approaches, or possibilities you've considered using to solve the same problem. -->

#### Testing
I loaded up the Doctor, Pharmacist, and Medical Resident professions, walked around a bit, and saw that they had the right items and proficiencies. Doctor and Pharmacist are tied to an achievement, so I did unlock all achievements to make them.
<!-- Describe what steps you took to test that this PR resolved the bug or added the feature, and what tests you performed to make sure it didn't cause any regressions.  Also, include testing suggestions for reviewers and maintainers. See TESTING_YOUR_CHANGES.md -->


<!--README: Cataclysm: Dark Days Ahead is released under the Creative Commons Attribution ShareAlike 3.0 license.
The code and content of the game are free to use, modify, and redistribute for any purpose whatsoever.
By contributing to the project you agree to the terms of the license and that any contribution you make will also be covered by the same license.
See http://creativecommons.org/licenses/by-sa/3.0/ for details. -->

theyareclose committed:
Adding Doctor and Pharmacist Professions

This adds a Doctor and Pharmacist profession. It also generalizes the "MD" trait to include more medical professionals.

theyareclose committed:
Apply suggestions from code review

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

Venera3 commented:
Pharmacists shouldn't get wound/burn care - their training unsurprisingly doesn't include too much hands-on time with wounds.

Just casually starting with a bit of morphine and a scalpel is pretty overblown for a GP as well and I'm not familiar enough with US regs regarding opiates in out-of-hospital settings to say if a pharmacist would get their hands on it but I'm doubtful.

theyareclose commented:
> Pharmacists shouldn't get wound/burn care - their training unsurprisingly doesn't include too much hands-on time with wounds.
> 
> Just casually starting with a bit of morphine and a scalpel is pretty overblown for a GP as well and I'm not familiar enough with US regs regarding opiates in out-of-hospital settings to say if a pharmacist would get their hands on it but I'm doubtful.

Yeah, you're right that the morphine really doesn't make sense for either. For pharmacists, there actually is morphine available in community pharmacies in the US, but it's usually in pill form (which isn't in CDDA afaik), so yes I will remove it for both of them.

I imagined for the scalpel that it could've been grabbed during evacuation. Since there had been some chaos up to that point, I figured a doctor might think to do it. Plus I think it's kind of fun... BUT if it seems that implausible, I can remove it.

Good call on the wound/burn care for the pharmacist, not sure where I got that idea. My partner is even a pharmacist, and apparently I overestimated their skills! I'll remove for sure. To make them more interesting, I'm going to add in some extra pharmaceuticals instead.

When testing, I noticed some mistakes I made in item quantities too, so I'm going to work on your changes and some item quantity fixes. Thanks for the feedback!

theyareclose committed:
Update professions.json

Incorporating feedback on pharmacist and morphine, fixing item amounts to spawn correctly, and adding more pharmaceuticals to the pharmacist.

theyareclose committed:
Apply suggestions from code review

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

theyareclose committed:
Update professions.json

Resolving conflict with aspirin in medical resident.
